### ISSUE TITLE
setuptools 18.5 says invalid version

### ISSUE THREAD

**Issue by johnyf** [tulip-control/polytope#18]
```
~/.virtualenvs/dev/lib/python2.7/site-packages/setuptools/dist.py:294: UserWarning: The version specified ('0.1.2-dev-93d397e7686c65df543f1cd40556ad438a6f8e3b') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
```

**Comment by johnyf** [comment]
In [`polytope.version`](https://github.com/tulip-control/polytope/blob/4db168890c950890b2e0bf66af3fccc3fe1585f7/polytope/version.py#L62) the changes in [`github.com/tulip-control/tulip-control/a0d1c0948ff8021ba48998eb1c52dfb578e3689c`](https://github.com/tulip-control/tulip-control/commit/a0d1c0948ff8021ba48998eb1c52dfb578e3689c) are needed.

**Comment by slivingston** [comment]
Is there any reason not to apply the patch directly?

We could also consider including a copy of the unit tests for version string generation. To avoid excessive duplication, I recommend that we manually track changes to version.py in the TuLiP repository (which should be rare). Version checking can also occur as part of integration tests.

**Comment by johnyf** [comment]
The only reason was to confirm that this was just not applied yet. Fixed in 4b3d1c3d28c344285557b4bbdff65aedb88a841a.

Admittedly, this duplication is characteristic of `setup.py`, due to its bootstrapping nature. Otherwise this baggage could have been moved to a common mini dependency, with its own tests at one place. It may be better to maintain a copy per project, to keep things local.

In the future, I plan to propose a top-down version definition (`setup.py` generating `version.py` automatically), as implemented in `dd`. That is somewhat simpler.

The remaining complex and duplicated part will be generation of a local version identifier that contains a `git` hash. In principle, this could be shifted to a common dependency, because it is mostly a development-time utility, when repeated installations should occur, in order for the local version identifier to be updated (unless the identifier is fetched dynamically upon package import, when installed as `dev`, though that is less desirable, for reasons to be discussed elsewhere).
